### ISSUE TITLE
Loading many R6 objects into environment causes RStudio console to hang for ~10mins

### ISSUE THREAD

**Issue by chwpearse** [rstudio/rstudio#4150]
### System details

RStudio Edition : Desktop
RStudio Version : 1.2.1233
OS Version      : Windows 10 home
R Version       :  3.5.1

### Steps to reproduce the problem

Run the following code. The execution time is < 1 sec but it takes around 10 minutes to load the object previews into the environment panel.
```{r}
library(R6)
TestObject <- R6::R6Class(
  public = list(),
  private = list()
)
start_time <- Sys.time()

for(i in 1:100){
  assign(paste0("A", i), TestObject)
}

end_time <- Sys.time() - start_time
end_time
```

### Describe the problem in detail

Having large numbers of R6 objects causes the object previews to load very slowly. While this is loading you cannot run code or open/save files.
This happens when you load an environment with objects in the .Rdata file, debug code using browse() statements or by pressing the refresh button in the top right of the environment pane.

I tested in RStudio version 1.1.423 duplicating an object as above and calculating end_time in a seperate console command immediately after. I recorded the following times

|objects | seconds|
|------|----|
100 | 643    
70 |  131.56 
60  |  70.75 
50   | 40.63 
40  |  20.12 
30  |   7.43 
20  |   2.53 
10   |  0.44 

`data.frame(objects = c(10*(1:7),100), time = c(0.44, 2.53, 7.43, 20.12, 40.63, 70.75, 131.56, 643))`

I also tested in Rstudio version 1.2.1233 and got these times

|objects | seconds|
|------|----|
10    | 0.44 
50   | 40.91 
100  | 502    

I tested in RStudio Version 0.99.903 and the problem is not present

|objects | seconds|
|------|----|
10    | 0.19 
100   | 0.38 
1000  | 1.27 

I tested again in RStudio version 1.1.423 but set TestObject$parent_env <- NULL after creating the object and recorded

|objects | seconds|
|------|----|
| 40  |  0.34 |
| 100 |  0.39  | 
| 1000 | 1.18  |

There is noticable delay when testing in R.exe console
The RStudio debug console showed a get_environment_state and a get_events running for the entire wait time
This suggests that newer versions of RStudio may be something when loading R6 objects into the environment preview that 

### Describe the behavior you expected

Loading ~100 simple R6 objects into an evironment in under a minute.

R version 3.5.1 (2018-07-02)
Platform: i386-w64-mingw32/i386 (32-bit)
Running under: Windows >= 8 x64 (build 9200)

Matrix products: default

locale:
[1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252   
[3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                      
[5] LC_TIME=English_Australia.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] R6_2.3.0

loaded via a namespace (and not attached):
[1] compiler_3.5.1 tools_3.5.1

**Comment by kevinushey** [comment]
We have a method that attempts to find external pointers (including NULL external pointers) which appears to be taking too long to run here:

``` r
> system.time(.Call("rs_hasExternalPointer", A1, TRUE))
   user  system elapsed 
  0.303   0.000   0.303 
```

Since that's being called for each R6 object in the environment when the Environment pane is populated, it eats up a lot of time.

It looks like most of the time is spent sorting the object names in the environment:

<img width="1182" alt="screen shot 2019-01-12 at 5 06 18 pm" src="https://user-images.githubusercontent.com/1976582/51080217-6a875f80-168c-11e9-84bc-47dc10db04b9.png">

**Comment by ronblum** [comment]
Verified fixed in 1.2.1242-3. Also, this causes 1.1.463 to hang when restarting and the variables are still in the environment (bonus repro).
